ott/geometry/low_rank.py

```diff
@@ -285,11 +290,20 @@ def _mask_subset_helper(
         aux_data, [c1, c2, src_mask, tgt_mask] + children
     )
 
+  def __add__(self, other: 'LRCGeometry') -> 'LRCGeometry':
```

Comment:
this is nice and elegant, but maybe add a simple comment here on what is done here, to facilitate readability for new contributors to the toolbox. Or maybe, better, at the top of the class description? This could be something like

"""Geometry whose cost is defined by product of two low-rank matrices.

Implements geometries that are defined as low rank products, i.e. for which
there exists two matrices :math:`A` and :math:`B` of `r` columns such that
the cost of the geometry equals :math:`AB^T`. Apart from being faster to
apply to a vector, these geometries are characterized by the fact that adding
two such geometries should be carried out by concatenating factors, i.e. 
if :math:`C=AB^T` and :math:`D=EF^T` then :math:`C+D=[A,E][B,F]^T`.

Reply:
Added this to the `__init__` description, thanks!